flamedisx/xenon/itp_map.py

```diff
@@ -0,0 +1,166 @@
+"""Code to load XENON-specific correction maps
+
+Adapted from https://github.com/XENONnT/straxen/blob/master/straxen/itp_map.py
+"""
+import logging
+import gzip
+import json
+import re
+
+import numpy as np
+from scipy.spatial import cKDTree
+
+import flamedisx as fd
+export, __all__ = fd.exporter()
+
+
+@export
+class InterpolateAndExtrapolate:
```

Comment:
Reminiscent of `InterpolateAndExtrapolate` and `InterpolatingMap` in xenon_corrections.py but with minor changes to `InterpolatingMap`.

Reply:
Indeed, this is the same code with minor changes (backported from straxen), moved to a different file